pkg/ostree/ostree.go

```diff
@@ -234,8 +240,30 @@ func Resolve(source SourceSpec) (CommitSpec, error) {
 		URL: source.URL,
 	}
 
+	if source.RHSM && source.MTLS != nil {
+		return commit, NewResolveRefError("cannot use both RHSM and MTLS when resolving ref")
+	}
+
 	if source.RHSM {
+		var subs *rhsm.Subscriptions
+		var err error
+
 		commit.Secrets = "org.osbuild.rhsm.consumer"
+		subs, err = rhsm.LoadSystemSubscriptions()
+
+		if err != nil {
+			return commit, NewResolveRefError("error adding rhsm certificates when resolving ref: %s", err)
+		}
+
+		if subs.Consumer == nil {
+			return commit, NewResolveRefError("error adding rhsm certificates when resolving ref")
+		}
+
+		source.MTLS = &MTLS{}
+		source.MTLS.ClientCert = subs.Consumer.ConsumerCert
+		source.MTLS.ClientKey = subs.Consumer.ConsumerKey
+	} else if source.MTLS != nil {
+		commit.Secrets = "org.osbuild.mtls"
```

Comment:
Would it make sense to make it an error when both MTLS and RHSM are set?  It's fine to have a documented priority like we do now, but maybe making it a hard error would be better for troubleshooting unexpected behaviour.

Reply:
This is already there see few lines above:

```
	if source.RHSM && source.MTLS != nil {
		return commit, NewResolveRefError("cannot use both RHSM and MTLS when resolving ref")
	}
```